apiserver/facades/client/applicationoffers/applicationoffers.go

```diff
@@ -166,19 +180,24 @@ func (api *OffersAPIv5) makeAddOfferArgsFromParams(user names.UserTag, backend B
 	if result.OfferName == "" {
 		result.OfferName = result.ApplicationName
 	}
-	application, err := backend.Application(addOfferParams.ApplicationName)
+
+	// If we have the full result, just return early.
```

Comment:
This return could be earlier right? You could just not even get the model database if you already have this. Though this may be premature optimisation, it is easier to read like this.